torch/_dynamo/variables/builder.py

```diff
@@ -540,6 +540,25 @@ def index_source(key):
                 source=self.source,
                 guards=make_guards(GuardBuilder.FUNCTION_MATCH),
             )
+        elif isinstance(value, types.MethodType):
```

Comment:
This graph breaks like crazy because of the nn module assumptions here - I forked the PR post discussion to https://github.com/pytorch/pytorch/pull/95353/files - the primary change is adding an nnmodule check - because there are lots of methods on other things that *should* work going through this, but are probably not good candidates for the implicit inlining this forces.

Reply:
ok i didn't see your comment until after i read your forked PR and cherry picked just this piece.  I wrote up a comment/TODO that aligns with what you said here. Thanks for your help figuring this out!